unity/unity-embed-host/UnsafeExtensions.cs

```diff
@@ -11,9 +16,44 @@ namespace Unity.CoreCLRHelpers;
 /// </summary>
 static class UnsafeExtensions
 {
+#if TESTING_UNITY_CORECLR
+    private static readonly ConcurrentBag<GCHandle> handles = new ();
+#endif
+
     public static object ToManagedRepresentation(this nint intPtr)
         => Unsafe.As<nint, object>(ref intPtr);
 
     public static nint ToNativeRepresentation(this object obj)
-        => Unsafe.As<object, nint>(ref obj);
+    {
+        var result = Unsafe.As<object, nint>(ref obj);
+#if TESTING_UNITY_CORECLR
```

Comment:
Nit, you may want to move `WorkaroundToGetGCSafety` first. As in theory the object could be moved after your cast but before the pin.